.github/workflows/python-package.yml

```diff
@@ -17,11 +17,16 @@ on:
   merge_group:
     types: [checks_requested]
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.head_ref }}
```

Comment:
Looks like it's possible to trigger a syntax error if not in a PR workflow. 
https://docs.github.com/en/actions/using-jobs/using-concurrency#example-using-a-fallback-value

And the current workflow does server more than pull_request.

```

on:
  push:
    branches: ['main']
    paths:
      - 'flaml/**'
      - 'test/**'
      - 'notebook/**'
      - '.github/workflows/python-package.yml'
      - 'setup.py'
  pull_request:
    branches: ['main']
  merge_group:
    types: [checks_requested]
```
```suggestion
  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.head_ref || github.run_id }}
```

Reply:
> And the current workflow does server more than pull_request.

Not sure if I understand what you mean...

Reply:
The workflow can be triggered by push to main, in that case it will raise error

Reply:
what kind of error? if a variable is not set GH gives empty string not error

Reply:
I didn't verify it. Just read it from the doc https://docs.github.com/en/actions/using-jobs/using-concurrency#example-using-a-fallback-value

Reply:
telling from experience as I have implemented this on other projects :chipmunk: